Source/santa-driver/SantaPrefixTree.h

```diff
@@ -0,0 +1,67 @@
+/// Copyright 2018 Google Inc. All rights reserved.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///    http://www.apache.org/licenses/LICENSE-2.0
+///
+///    Unless required by applicable law or agreed to in writing, software
+///    distributed under the License is distributed on an "AS IS" BASIS,
+///    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+///    See the License for the specific language governing permissions and
+///    limitations under the License.
+
+#ifndef SANTA__SANTA_DRIVER__SANTAPREFIXTREE_H
+#define SANTA__SANTA_DRIVER__SANTAPREFIXTREE_H
+
+#include <libkern/c++/OSDictionary.h>
+#include <libkern/locks.h>
+
+///
+///  SantaPrefixNode is wrapper class representing one char in a prefix.
+///
+///  OSDefineMetaClassAndStructors is picky so this can't be a nested class.
+///
+class SantaPrefixNode : public OSObject {
+  OSDeclareDefaultStructors(SantaPrefixNode);
+
+ public:
+  bool init() override;
+  void free() override;
+
+  bool isPrefix;
+  // TODO(bur): After writing this I noticed OSDictionary has an O(n) runtime.
+  //            Switch to an int[256] lookup table and get rid of SantaPrefixNode.
```

Comment:
I'd do this to start with as changing away from OSDictionary allows a bunch of simplifications elsewhere (you can nest the classes, do away with the OSDeclare* macros, use single chars instead of 'string' keys, etc.)

Reply:
BTW, if going this route, using half-byte keys instead of chars would reduce the space requirements by a factor of 8, because each level then only needs an array of 2^4 = 16 pointers instead of 2^8 = 256 pointers, while the effective length of the prefix doubles.